pootle/apps/pootle_data/utils.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils.functional import cached_property
+
+from pootle_statistics.models import SubmissionTypes
+from pootle_store.constants import FUZZY, OBSOLETE, TRANSLATED, UNTRANSLATED
+from pootle_store.models import Suggestion
+from pootle_store.util import SuggestionStates
+
+
+User = get_user_model()
+
+
+class StoreDataTool(object):
+    """Sets the stats for a store"""
+
+    def __init__(self, store):
+        self.store = store
+
+    @property
+    def units(self):
+        return self.store.unit_set.live()
+
+    @cached_property
+    def last_created_unit(self):
+        """Last unit created in this Store"""
+        return (
+            self.units.exclude(creation_time__isnull=True)
+                      .order_by('-creation_time', '-revision').first())
+
+    @cached_property
+    def last_updated_unit(self):
+        """Last unit updated in this Store"""
+        return self.units.order_by('-mtime', '-revision').first()
+
+    @cached_property
+    def last_submission(self):
+        """Last non-UNIT_CREATE submisison for this store"""
+        return (
+            self.store.submission_set.exclude(type=SubmissionTypes.UNIT_CREATE)
+                                     .latest())
+
+    @cached_property
+    def wordcount(self):
+        wordcount = (
+            self.units.order_by().values_list('state')
+                      .annotate(wordcount=models.Sum('source_wordcount')))
+        result = dict(total=0, translated=0, fuzzy=0)
+        for state, wordcount in wordcount.iterator():
+            result['total'] += wordcount
+            if state == TRANSLATED:
+                result['translated'] = wordcount
+            elif state == FUZZY:
+                result['fuzzy'] = wordcount
+        return result
+
+    @property
+    def suggestion_count(self):
+        return Suggestion.objects.filter(
+            unit__store=self,
+            unit__state__gt=OBSOLETE,
+            state=SuggestionStates.PENDING).count()
```

Comment:
When I see stuff like that I always wonder why not using `self.store.suggestion_set` and thus drop the `Suggestion` import?